dtfs-central-api/src/cron-scheduler-jobs/deal-cancellation/cancel-deal-job.ts

```diff
@@ -0,0 +1,39 @@
+import { asString, CronSchedulerJob, TfmDeal } from '@ukef/dtfs2-common';
+import { generateSystemAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { endOfDay } from 'date-fns';
+import { TfmDealCancellationRepo } from '../../repositories/tfm-deals-repo';
+
+const { DEAL_CANCELLATION_SCHEDULE } = process.env;
+
+/**
+ * Updates deals to be cancelled
+ * @param dealIds the deals to be cancelled
+ */
+const cancelDeals = async (deals: TfmDeal[]) => {
+  await Promise.all(deals.map((deal) => TfmDealCancellationRepo.submitDealCancellation(deal._id, deal.tfm.cancellation!, generateSystemAuditDetails())));
```

Comment:
Just wondering if we need to add `non-null assertion` operator?

Reply:
Should we check whether deal is already cancelled?

Reply:
resolving conversation since approved so can merge in :)